tutorial/doppleganger.js

```diff
@@ -0,0 +1,529 @@
+//======================================
+// Version 1.0
+//  doppleganger.js
+//
+//  Created by Timothy Dedischew on 04/21/2017.
+//  Copyright 2017 High Fidelity, Inc.
+//  Copyright 2022 Overte e.V.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+//
+/* global module */
+// @module doppleganger
+//
+// This module contains the `Doppleganger` class implementation for creating an inspectable replica of
+// an Avatar (as a model directly in front of and facing them).  Joint positions and rotations are copied
+// over in an update thread, so that the model automatically mirrors the Avatar's joint movements.
+// An Avatar can then for example walk around "themselves" and examine from the back, etc.
+//
+// This should be helpful for inspecting your own look and debugging avatars, etc.
+//
+// The doppleganger is created as an overlay so that others do not see it -- and this also allows for the
+// highest possible update rate when keeping joint data in sync.
+//
+// NOTE: THIS IS A MODIFIED VERSION SPECIFICALLY FOR THE TUTORIAL.
+
+module.exports = Doppleganger;
+
+// @property {bool} - when set true, Script.update will be used instead of setInterval for syncing joint data
+Doppleganger.USE_SCRIPT_UPDATE = false;
+
+// @property {int} - the frame rate to target when using setInterval for joint updates
+Doppleganger.TARGET_FPS = 60;
+
+// @property {int} - the maximum time in seconds to wait for the model overlay to finish loading
+Doppleganger.MAX_WAIT_SECS = 10;
+
+// @function - derive mirrored joint names from a list of regular joint names
+// @param {Array} - list of joint names to mirror
+// @return {Array} - list of mirrored joint names (note: entries for non-mirrored joints will be `undefined`)
+
+var Setarry; 
+
+Doppleganger.getMirroredJointNames = function(jointNames) {
+    return jointNames.map(function(name, i) {
+        if (/Left/.test(name)) {
+            return name.replace('Left', 'Right');
+        }
+        if (/Right/.test(name)) {
+            return name.replace('Right', 'Left');
+        }
+        return undefined;
+    });
+};
+
+// @class Doppleganger - Creates a new instance of a Doppleganger.
+// @param {Avatar} [options.avatar=MyAvatar] - Avatar used to retrieve position and joint data.
+// @param {bool}   [options.mirrored=true]   - Apply "symmetric mirroring" of Left/Right joints.
+// @param {bool}   [options.autoUpdate=true] - Automatically sync joint data.
+function Doppleganger(options) {
+    options = options || {};
+    this.avatar = options.avatar || MyAvatar;
+    this.mirrored = 'mirrored' in options ? options.mirrored : false;
+    this.autoUpdate = 'autoUpdate' in options ? options.autoUpdate : true;
+
+    // @public
+    this.active = false;   // whether doppleganger is currently being displayed/updated
+    this.entityID = null; // current doppleganger's Entity id
```

Comment:
```suggestion
    this.entityID = null;  // current doppleganger's Entity id
```
Just to bring it in line with the other comments.